apps/prairielearn/src/pages/instructorCourseAdminSettings/instructorCourseAdminSettings.html.ts

```diff
@@ -18,111 +18,172 @@ export function InstructorCourseAdminSettings({
+                  disabled
+                />
+                <small class="form-text text-muted">
+                  The allowable timezones are from the tz database. It's best to use a city-based
+                  timezone that has the same times as you.
+                </small>
+              </div>
+              <div class="form-group">
+                <label for="path">Path</label>
+                <input
+                  type="text"
+                  class="form-control"
+                  id="path"
+                  name="path"
+                  value="${resLocals.course.path}"
+                  disabled
+                />
+                <small class="form-text text-muted">
+                  The path where course files can be found.
+                </small>
+              </div>
+              <div class="form-group">
+                <label for="repository">Repository</label>
+                <input
+                  type="text"
+                  class="form-control"
+                  id="repository"
+                  name="repository"
+                  value="${resLocals.course.repository}"
+                  disabled
+                />
+                <small class="form-text text-muted">
+                  The Github repository that can be used to sync course files.
+                </small>
+              </div>
+              <p class="mb-0">
+                ${coursePathExists && courseInfoExists
+                  ? resLocals.authz_data.has_course_permission_view
+                    ? resLocals.authz_data.has_course_permission_edit &&
+                      !resLocals.course.example_course
+                      ? html`
+                          <a
+                            data-testid="edit-course-configuration-link"
+                            href="${resLocals.urlPrefix}/${resLocals.navPage}/file_edit/infoCourse.json"
+                          >
+                            Edit course configuration
+                          </a>
+                          in <code>infoCourse.json</code>
+                        `
+                      : html`
+                          <a
+                            href="${resLocals.urlPrefix}/${resLocals.navPage}/file_view/infoCourse.json"
+                          >
+                            View course configuration
+                          </a>
+                          in <code>infoCourse.json</code>
+                        `
+                    : ''
+                  : ''}
+              </p>
+            </form>
           </div>
         </main>
       </body>
     </html>
   `.toString();
 }
+
+function CourseDirectoryMissingAlert({
+  resLocals,
+  coursePathExists,
+  courseInfoExists,
+}: {
+  resLocals: Record<string, any>;
+  coursePathExists: boolean;
+  courseInfoExists: boolean;
+}) {
+  if (resLocals.authz_data.has_course_permission_edit && !resLocals.course.example_course) {
```

Comment:
There's a good application of [De Morgan's law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) here: you can invert the conditional and use an early return.

```ts
if (!resLocals.authz_data.has_course_permission_edit || resLocals.course.example_course) {
  return '';
}
```

Reply:
Inverted the conditional here for an early return in [8e3b7a9](https://github.com/PrairieLearn/PrairieLearn/pull/9386/commits/8e3b7a92c2e5562595292f74ea7fb09cb6932da6).